[chore] add spot hover menu

Kharonus opened the pull request:
- added a hover menu container, which can be inserted into any `spot-list--item-action`
- is positioned absolute to the parent

Hint: be aware, that this branch merges the `spot-list_compact` by @b12f . Hence, the real diff is only visible, when #10748 is merged.

Commit:
Add compact version of spot-list

This adds the modifier `spot-list_compact`, which compresses `spot-list`
vertically by reducing paddings. It also increases the padding on the
right hand side of items, since it differed from the left.

Commit:
Update spot-list hover color and compact spacings

Kharonus committed:
Merge remote-tracking branch 'origin/feature/spot-list_compact' into chore/add-spot-hover-menu

Kharonus committed:
[chore] new hover menu for spot-list

b12f commented on frontend/src/app/spot/styles/sass/components/list.sass:
```diff
@@ -13,22 +13,42 @@
         color: $spot-color-basic-gray-3
         cursor: default
 
+    &-hover
```
I would call this `hover-actions` or `actions`. Maybe @marcalcobe or @psatyal have an opinion, too.

HDinger replied:
I agree. `Hover` is an activity not an object that can be selected or styled

Kharonus replied:
currently it results into an html like this:

```
<div class="spot-list--item-title" ...
<div class="spot-list--item-hover" ...
```

meaning, that the hover is defined right next to the content. As the parent is already called `spot-list--item-action` only `actions` seems weird. but I could fully go with `hover-actions`. Will talk to product team about the domain speech and how they would call this very piece of UI.

Kharonus replied:
After speaking with parimal, we agreed to adapt the name from the figma files, which is `-floating-actions`. This kind of decouples it from the trigger, which is hover, and emphasizes the actual behaviour of the component: floating over the sibling content.

b12f commented on frontend/src/app/spot/styles/sass/components/list.sass:
```diff
@@ -13,22 +13,42 @@
+        > .spot-icon
+          width: 1.125rem
+          height: 1.125rem
+          font-size: 0.75rem
```
Sure we need these? Maybe we can change the design slightly and remove this

HDinger replied:
I stumbled over that as well. I guess the only reason for that is to reduce the spacing, right? The big spacing is a result of [this line](https://github.com/opf/openproject/pull/10762/files#diff-254007e79e18a369555f7a70720e24a63882fd51e6607eed9713119b2169eea1R45). So maybe we can simply drop all of that? 
In any case, please use the `spot-spacing-..` variables at least for the font-size

Kharonus replied:
this is a very valid question. I was asking myself, why we have here the only occurance of 18px icons in the whole styles. Will ask @psatyal .

Kharonus replied:
Parimal agrees to our confusion. For now, we documented in figma, that it is weird to have this exceptional size for icon buttons there (18px appears to be not used anywhere else), and we removed the exeception in code. so, now the floating action buttons are 24px width and height.

Kharonus committed:
[chore] removed exception and renamed menu

Kharonus commented:
@b12f  and @HDinger 

reworked after consulting @psatyal

Kharonus committed:
[chore] introduced floating wrapper

- enable correct hover effects on list items with flaoting actions
- align children to top when list item gets line breaks

Kharonus committed:
[chore] moved comment to paragraph

HDinger commented on frontend/src/app/spot/styles/sass/components/list.sass:
```diff
@@ -14,32 +14,68 @@
+      right: 0
+      height: 100%
+      padding: $spot-spacing-0_5 $spot-spacing-1
+      background: linear-gradient(90deg, rgba($spot-color-basic-gray-6, 0) 0%, $spot-color-basic-gray-6 11.98%)
```
11.98% is a very random value.. can't it be 12%?

Kharonus replied:
it seems random and was directly taken from the figma sketches. Not sure, what's the background of this value, @psatyal ?

psatyal replied:
Hi @HDinger and @Kharonus.

Could you show me where the 11,98% is in Figma? I can't find it when I look (it's 100% to 0%, but since @marcalcobe worked on this, perhaps it's elsewhere and I don't know).

psatyal replied:
As for mobile, yes, we'll have to work on describing mobile behaviour if that's not done yet. It would surely involve something a dots menu, as @Kharonus suggested in [another comment ](https://github.com/opf/openproject/pull/10762#issuecomment-1149870629).

Kharonus replied:
11.98% can be switched to `1rem`. Talked to parimal about the behaviour for the fading background of the floating actions. And it makes no sense to create them dependent of the width of the floating actions container.
Instead it should be a fixed spacing width.

I have this fixed in the subsequent PR #10658, but if you want, I can cherry pick it and commit it here, too.

HDinger reviewed (commented):
* When I have a very long word, the text continues on the right side of the menu:
<img width="898" alt="Bildschirmfoto 2022-06-08 um 14 23 31" src="https://user-images.githubusercontent.com/7457313/172615909-ca1fdabb-7f50-407e-869b-d64c23582dda.png">

* In order to see the menu on mobile, I have to click on the label. However this automatically selects the checkbox. I can imagine that this will cause problems when the hierarchy is automatically selected as well. Thus I would accidentally select the whole hierarchy and may loose information if I have selected parts of it beforehand.

Kharonus commented:
@HDinger 
Long words without space won't get wrapped, so question is, if we should put `overflow: hidden` on `spot-list--item-action`. @b12f any opinion on that?

To the mobile issue: This is still completely uncovered. We have no alternative concept for the hover on the list. I would probably not try to make it somehow work, but instead replace it with a mobile ready concept like a dots menu on mobile resolutions. But this is not yet decided by product team, AFAIK. @psatyal , any words to that topic?

Kharonus committed:
Merge branch 'dev' into chore/add-spot-hover-menu

Kharonus committed:
[chore] reworked spot list item style

Kharonus committed:
[chore] add word wrap to the spot list title

HDinger reviewed (approved):
👍
